citesphere/src/main/java/edu/asu/diging/citesphere/web/user/AsyncCitationManager.java

```diff
@@ -86,8 +100,66 @@ public AsyncUpdateCitationsResponse getUpdateCitationsResponse(String taskId)
      * @param taskId id of the task that is not needed in memory
      */
-    public void clearTask(String taskId) {
-        taskTracker.remove(taskId);
+    public void clearUpdateTask(String taskId) {
+        updateTaskTracker.remove(taskId);
+    }
+
+    /**
+     * Deletes Citations asynchronously
+     * 
+     * @param user           User accessing Zotero
+     * @param groupId        GroupId of the Citations
+     * @param citationIdList List of the Citations to be deleted
+     * @return AsyncDeleteCitationsResponse containing taskId and its current status
+     */
+    public AsyncDeleteCitationsResponse deleteCitations(IUser user, String groupId, List<String> citationIdList)
+            throws ZoteroConnectionException, ZoteroHttpStatusException {
+        String taskId = UUID.randomUUID().toString();
+        Future<Map<ItemDeletionResponse, List<String>>> futureTask = asyncDeleteCitationsProcessor.deleteCitations(user,
+                groupId, citationIdList);
+        deleteTaskTracker.put(taskId, futureTask);
+        AsyncDeleteCitationsResponse asyncResponse = new AsyncDeleteCitationsResponse();
+        asyncResponse.setTaskID(taskId);
+        asyncResponse.setTaskStatus(AsyncTaskStatus.PENDING);
+        return asyncResponse;
+    }
+
+    /**
+     * Gets the current status of the deletion task. If the task is completed, it
+     * returns the status for each Citation which was requested to be deleted.
+     * 
+     * @param taskId Id of the task
+     * @return AsyncDeleteCitationsResponse containing task id, task status, and
+     *         response if the task is completed
+     */
+    public AsyncDeleteCitationsResponse getDeleteCitationsResponse(String taskId)
+            throws ExecutionException, InterruptedException {
+        AsyncDeleteCitationsResponse response = new AsyncDeleteCitationsResponse();
+        Future<Map<ItemDeletionResponse, List<String>>> futureTask = deleteTaskTracker.get(taskId);
+        response.setTaskID(taskId);
+        if (futureTask.isDone()) {
+            response.setTaskStatus(AsyncTaskStatus.COMPLETE);
+            response.setResponse(futureTask.get());
+        } else {
+            response.setTaskStatus(AsyncTaskStatus.PENDING);
+        }
+        return response;
+    }
+
+    /**
+     * Removes the Citation deletion task from memory and removes the deleted
+     * Citations from index and local database
+     * 
+     * @param taskId Id of the task
+     */
+    public void clearDeleteTask(String taskId) throws InterruptedException, ExecutionException {
+        Map<ItemDeletionResponse, List<String>> deletionResponse = deleteTaskTracker.remove(taskId).get();
+        for (String itemKey : deletionResponse.getOrDefault(ItemDeletionResponse.SUCCESS, new ArrayList<>())) {
+            Optional<ICitation> citation = citationStore.findById(itemKey);
+            if (citation.isPresent()) {
+                citationStore.delete(citation.get());
+            }
+        }
```

Comment:
removing the citations from the local db shouldn't be done in the clear method. The clear method should just do what it says, clearing the task from the local cache. Especially since this method is called from a controller, what if the user closes the browser before the clear method can ever be called? Return a future implementation that allows you to add a callback method to this.